app/models/service.rb

```diff
@@ -71,6 +70,9 @@ class Service < ApplicationRecord
   include_concern 'Aggregation'
   include_concern 'ResourceLinking'
 
+  virtual_total :v_total_vms, :vms,
+                :arel => aggregate_hardware_arel("v_total_vms", vms_tbl[:id].count, :skip_hardware => true)
```

Comment:
I'm fine with this, but I'm curious why we would need an arel, when we already have virtual_total which is a specialized method to begin with.  Isn't it like specializing a specialized method?

Reply:
This would work if `:vms` was an actual relation on service, but it is not.  It is another `virtual_attribute`, which is an alias to another `virtual_attribute`.  forgetting some of the details, but there is also issues with the fact that this is a polymorphic relationship with VMs, and I don't think that `virtual_total` is built with that in mind.